### PR TITLE
feat: allow launching more games

### LINKED ISSUE #7: Support more games
See https://github.com/spicylobstergames/SpicyLauncher#supported-games

- [ ]  CLI
- [ ] GUI (needs update in design)

### PR DESCRIPTION
**Pull request by orhun**
closes #7

### CODE REVIEW

**Comment by zicklag** [comment]
Hey @orhun _big_ apology for merging prematurely on my previous commits.

The `#launcher` channel on the Revolt bridge seems to be flaking out so I'm not sure if you got my messages there.

Anyway, I see that you reverted my commits and opened this to get the proper review. 👍

I absolutely don't want to take over or anything like that, and I'm really sorry for the inconvenience. I admit that I rushed merging that and that was a mistake.

Do you want me to open a new PR so that I can address some of the comments you made:

- remove debug prints (console.log) in the code.
- use the custom-defined error types (error.rs) instead of defining your own struct
- use the Display implementation instead of Game::id
    - Though I believe that `Display` would be more suitable for replacing `Game::name()` instead, though, right? Because the name represents the display version of the game, not the machine ID.
- there are formatting issues in the code
    - I'm not sure what formatting issues you were referring to, so I'd need a pointer on that.

Or do you want me to push a commit to this branch to make those changes?

File: gui/src/components/GameTabs.svelte
Comment (bufgix) [inline gui/src/components/GameTabs.svelte:4]: Can you remove all console logs?

File: gui/src/components/GameTabs.svelte
Comment (bufgix) [inline gui/src/components/GameTabs.svelte:7]: We should remove this.

File: gui/src/components/GameTabs.svelte
Comment (bufgix) [inline gui/src/components/GameTabs.svelte:16]: it would be nice if we use [strict equal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality)(`===`) here and other places

File: gui/src/components/Sidebar.svelte
Comment (bufgix) [inline gui/src/components/Sidebar.svelte:41]: it would be nice if we create Game object as contant

```ts
// constants.ts
//...
export const GAMES = {
    JUMPY: 'Jumpy',
    PUNCHY: 'Punchy'
} 
```

and we can use `GAMES` key as type

```ts
// global.d.ts

export type Game = keyof typeof GAMES; // "JUMPY" | "PUNCHY"

```

So we can write in a simpler way

```suggestion
  $: {
       gameTitle = GAMES[$currentGame]
  }
```

**Review by bufgix — CHANGES_REQUESTED** [review]
Thanks for your contribution

**Comment by orhun** [comment]
> Hey @orhun _big_ apology for merging prematurely on my previous commits.
>
> Anyway, I see that you reverted my commits and opened this to get the proper review. +1
> 
> I absolutely don't want to take over or anything like that, and I'm really sorry for the inconvenience. I admit that I rushed merging that and that was a mistake.

No problem!

> Or do you want me to push a commit to this branch to make those changes?
> 
> Or do you just want me to hold off until we get @bufgix's review? Just let me know what you want!

You can simply apply changes here after the review and we can merge this PR. I see that @bufgix already submitted a review and I will also submit one soon.

Thanks for your understanding and cooperating on this ❤️

File: .github/workflows/cron.yml
Comment (orhun) [inline .github/workflows/cron.yml:20]: Is this line necessary? We are overwriting these files anyways.

File: .github/workflows/cron.yml
Comment (orhun) [inline .github/workflows/cron.yml:24]: Let's have a loop here in case we add more games in the future:

```sh
for game in "jumpy" "punchy"; do
  curl -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
    "https://api.github.com/repos/fishfolk/${game}/releases" > "${game}.json"
done
```

File: .github/workflows/cron.yml
Comment (orhun) [inline .github/workflows/cron.yml:32]:
```suggestion
          git commit -m "Update releases"
```

File: core/src/archive/zip.rs
Comment (orhun) [inline core/src/archive/zip.rs:16]: What's the reason for this change?

File: core/src/github/mod.rs
Comment (orhun) [inline core/src/github/mod.rs:14]: Please remove this function.

File: core/src/lib.rs
Comment (orhun) [inline core/src/lib.rs:2]: Can we remove this newline between imports?

File: core/src/lib.rs
Comment (orhun) [inline core/src/lib.rs:24]: This can be a part of `Error` struct. There is no need to define another type for error handling.

File: core/src/lib.rs
Comment (orhun) [inline core/src/lib.rs:14]:
```suggestion
/// The different games supported by the launcher.
```

File: core/src/lib.rs
Comment (orhun) [inline core/src/lib.rs:51]: Let's use `Display` implementation instead of having a `name` function. It feels more suitable that way.

https://doc.rust-lang.org/rust-by-example/hello/print/print_display.html

File: core/src/lib.rs
Comment (orhun) [inline core/src/lib.rs:62]: Not a big deal but can be rewritten as:

```rs
    pub fn binary_name(&self) -> String {
        let id = self.id();
	if cfg!(target_os = "windows") {
            format!("{id}.exe")
        } else {
            id.to_string()
        }
    }
```

I'm okay with both. I leave it up to you.

File: core/src/storage.rs
Comment (orhun) [inline core/src/storage.rs:85]: Please remove this newline.

File: core/src/storage.rs
Comment (orhun) [inline core/src/storage.rs:89]: Please remove this newline.

File: core/src/storage.rs
Comment (orhun) [inline core/src/storage.rs:87]: `Vec::new()` or `vec![]` is more explanatory than `Vec::default()`. Let's change this to one of them.

File: .github/workflows/cron.yml
Comment (zicklag) [reply .github/workflows/cron.yml:20]: I put that there because I figured we should remove the existing `releases.json` file. We could just manually push a commit to remove that, though, so we could take it out.

File: core/src/archive/zip.rs
Comment (zicklag) [reply core/src/archive/zip.rs:16]: Now that we are putting the releases for the games an extra level deep in the data dir, the directory to extract the releases to may not have already been created.

For example, at startup we ensure the data dir `~/.SpicyLauncher` exists, and that was where we would extract the releases.

Now, with multiple games, though, we need to extract to `~/.SpicyLauncher/jumpy` or another game-specific folder which may not exist yet.

Alternatively we could make sure directories for each game exist before hand.

File: core/src/lib.rs
Comment (zicklag) [reply core/src/lib.rs:62]: Yeah, yours is better.

I learned while working on WGPU that using `cfg!()` is better sometimes because compiling on one platform will still catch type checker errors for the other platform, but `#[cfg()]` checks will not at all check the other platform code for correctness.

**Comment by zicklag** [comment]
I believe I addressed all the review points. Let me know if there's anything else!

**Pull request merged**
